File: WebApplication/ClientApp/src/app/_orchestration/getDashboardInfo/dashboardInfoRequestDto.ts
import { removeUndefinedKeyValues } from "../../_common/jsonHelpers"

export enum DashboardInfoType {
  Events,
  Athletes,
}

export enum DashboardInfoLocationType {
  All,
  State,
  Area,
  City
}

export class DashboardInfoRequestDto {

  constructor(
    private dashboardInfoType: DashboardInfoType,
    private dashboardInfoLocationType: DashboardInfoLocationType,
    private locationTermUrlFriendly: string | null = null,
  ) {}

  public getAsParamsObject = () => {

    const params = {
      dashboardInfoType: this.dashboardInfoType,
      dashboardInfoLocationType: this.dashboardInfoLocationType,
      locationTermUrlFriendly: this.locationTermUrlFriendly,
    }

    return removeUndefinedKeyValues(params)
  }
}

File: Orchestration/GetDashboardInfo/DashboardInfoRequestDto.cs

namespace Orchestration.GetDashboardInfo;

/// <summary>
/// Events or Athletes
/// </summary>
public enum DashboardInfoType
{
    Events,
    Athletes
}

/// <summary>
/// All, State, Area, or City
/// </summary>
public enum DashboardInfoLocationType
{
    All,
    State,
    Area,
    City
}

public class DashboardInfoRequestDto
{
    public DashboardInfoType DashboardInfoType { get; set; }

    public DashboardInfoLocationType DashboardInfoLocationType { get; set; }

    /// <summary>
    /// colorado, san-diego, greater-miami-area, etc. are examples. Needs to come in with dashes for spaces and lowercase.
    /// </summary>
    public string? LocationTermUrlFriendly { get; set; }
}
